File: gestaolegal/common/constants/user_roles.py
from enum import StrEnum


class UserRole(StrEnum):
    USER = "user"
    ADMINISTRADOR = "admin"
    ORIENTADOR = "orient"
    COLAB_PROJETO = "colab_proj"
    ESTAGIARIO_DIREITO = "estag_direito"
    COLAB_EXTERNO = "colab_ext"
    PROFESSOR = "prof"

    @classmethod
    def choices(cls) -> list[tuple[str, str]]:
        return [(role.value, role.display_name) for role in UserRole]

    @property
    def display_name(self) -> str:
        """Return human-readable display name for the role."""
        display_names = {
            "user": "Usuário",
            "admin": "Administrador",
            "orient": "Orientador",
            "colab_proj": "Colaborador de projeto",
            "estag_direito": "Estagiário de Direito",
            "colab_ext": "Colaborador externo",
            "prof": "Professor",
        }
        return display_names.get(self.value, self.value)

    @classmethod
    def get_display_name(cls, role_value: str) -> str:
        """Get display name for a role value string."""
        try:
            role = cls(role_value)
            return role.display_name
        except ValueError:
            return role_value

File: web/src/lib/constants/user-roles.ts
export const USER_ROLES = {
	ADMIN: 'admin',
	ORIENT: 'orient',
	COLAB_PROJ: 'colab_proj',
	ESTAG_DIREITO: 'estag_direito',
	COLAB_EXT: 'colab_ext',
	PROF: 'prof'
} as const;

export const USER_ROLE_VALUES = Object.values(USER_ROLES);

export type UserRole = (typeof USER_ROLES)[keyof typeof USER_ROLES];

export const USER_ROLE_OPTIONS = [
	{ value: USER_ROLES.ADMIN, label: 'Administrador' },
	{ value: USER_ROLES.ORIENT, label: 'Orientador' },
	{ value: USER_ROLES.COLAB_PROJ, label: 'Colaborador de projeto' },
	{ value: USER_ROLES.ESTAG_DIREITO, label: 'Estagiário de Direito' },
	{ value: USER_ROLES.COLAB_EXT, label: 'Colaborador externo' },
	{ value: USER_ROLES.PROF, label: 'Professor' }
] as const;

export const USER_ROLE_BADGE_MAP = {
	[USER_ROLES.ADMIN]: {
		text: 'Administrador',
		variant: 'secondary' as const,
		class: 'bg-primary text-primary-foreground border-primary'
	},
	[USER_ROLES.ORIENT]: {
		text: 'Orientador',
		variant: 'secondary' as const,
		class: 'bg-chart-1 text-primary-foreground border-chart-1'
	},
	[USER_ROLES.COLAB_PROJ]: {
		text: 'Colaborador de projeto',
		variant: 'secondary' as const,
		class: 'bg-chart-2 text-primary-foreground border-chart-2'
	},
	[USER_ROLES.ESTAG_DIREITO]: {
		text: 'Estagiário de Direito',
		variant: 'secondary' as const,
		class: 'bg-chart-3 text-primary-foreground border-chart-3'
	},
	[USER_ROLES.COLAB_EXT]: {
		text: 'Colaborador externo',
		variant: 'secondary' as const,
		class: 'bg-chart-4 text-primary-foreground border-chart-4'
	},
	[USER_ROLES.PROF]: {
		text: 'Professor',
		variant: 'secondary' as const,
		class: 'bg-chart-5 text-primary-foreground border-chart-5'
	}
} as const;

export function getUserRoleLabel(roleValue: string): string {
	const role = USER_ROLE_OPTIONS.find((r) => r.value === roleValue);
	return role?.label || roleValue;
}
